orderBook.js

```diff
@@ -0,0 +1,35 @@
+const reconcileOrder = (existingBook, incomingOrder)=> {
+  let updateBlock = []
+  
+  while(existingBook.length && incomingOrder.quantity > 0) {
+    const currentOrder = existingBook.shift()
+
+    if(currentOrder.type !== incomingOrder.type && currentOrder.price === incomingOrder.price)  {
+      if(incomingOrder.quantity === currentOrder.quantity) {
+        incomingOrder.quantity = 0
+      } else if (incomingOrder.quantity> currentOrder.quantity) {
+        incomingOrder.quantity = incomingOrder.quantity - currentOrder.quantity
+      } else {
+        incomingOrder = {
+          type: currentOrder.type,
+          price: currentOrder.price,
+          quantity: currentOrder.quantity - incomingOrder.quantity
+        }
+      }
+    } else {
+      updateBlock.push(currentOrder)
+    }
+  }
```

Comment:
This could be broken down better with helper functions. Having a function to determine match and another to actually fulfill the order would make this a lot cleaner. Remember functions should do one thing. Let reconcileOrder control flow of control and break the rest of the responsibilities out.